cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -679,6 +688,13 @@ func (c serviceAccountTokenControllerStarter) startServiceAccountTokenController
 	// start the first set of informers now so that other controllers can start
 	controllerContext.InformerFactory.Start(ctx.Done())
 
+	// register podIndexer here to allow it to be started later.
+	if legacyTokenCleanUp {
+		if err := controller.AddPodIndexer(controllerContext.InformerFactory.Core().V1().Pods().Informer().GetIndexer()); err != nil {
+			return nil, true, fmt.Errorf("error adding pod indexer to Tokens controller: %v", err)
+		}
```

Comment:
I'm trying to decide if watching all updates to all pods and maintaining a secrets-in-use index at all times is worth it for a cleanup we'll do ~daily, starting a ~year in the future, or if we should just hold a pod lister and do a point-in-time check of secret use within a namespace prior to cleaning up a secrets in that namespace

Reply:
Yeah. It's a trade-off. I can remove the secrets-in-use index and just keep the pod lister.